matchms/filtering/add_retention.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any
+from typing import List
+from matchms.utils import get_first_common_element
+from ..typing import SpectrumType
+
+
+_retention_time_keys = ["retention_time", "retentiontime", "rt", "scan_start_time"]
+_retention_index_keys = ["retention_index", "retentionindex", "ri"]
+
+
+def safe_store_value(spectrum: SpectrumType, value: Any, target_key: str) -> SpectrumType:
+    """Helper function to safely store a value in the target key without throwing an exception, but storing 'None' instead.
+
+    Args:
+        spectrum (SpectrumType): Spectrum to which to add 'value' in 'target_key'.
+        value (Any): Value to parse into 'target_key'.
+        target_key (str): Name of the key in which to store the value.
+
+    Returns:
+        SpectrumType: Spectrum with added key.
+    """
+    if value is not None:   # one of accepted keys is present
+        try:
+            value = float(value)
+            rt = value if value >= 0 else None  # discard negative RT values
+        except ValueError:
+            print("%s can't be converted to float.", value)
+            rt = None
+        spectrum.set(target_key, rt)
+    return spectrum
+
+
+def add_retention(spectrum: SpectrumType, target_key: str, accepted_keys: List[str]) -> SpectrumType:

```

Comment:
This is a helper function for `add_retention_time()`, right? We could add an underscore to indicate that (feel free to ignore though).

Reply:
Resolved in https://github.com/matchms/matchms/pull/265/commits/42f0a75c71faef8479dbd6fc2e37f2c560f25bef